### PR TITLE
Users/rtrufelea/complex recipe filtering

### PR DESCRIPTION
**Pull request by roxanatrufelea**
# Justification

TODO: #59

# Implementation

TODO: filter recipes by many fields and operations

# Testing

TODO: tested using swagger: [http://localhost:8080/kitchen/swagger-ui/#/recipe-controller/complexFilterRecipesUsingPOST](url)

# Checklist

- [ ] I tested changes to product code in product
- [ ] I considered updates to the README
- [ ] I considered updates to the error code dictionary

### CODE REVIEW

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/Model/Request/OperationType.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.kitchen.iChef.Controller.Model.Request;
+
+public enum OperationType {
+ greaterThan,lessThan,equals,greaterThanOrEqualTo,lessThanOrEqualTo,like
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Controller/Model/Request/OperationType.java:4]: one operation per line

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeFilterCriteria.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.kitchen.iChef.Repository;
+
+import com.kitchen.iChef.Controller.Model.Request.FilterRequest;
+
+import java.util.List;
+
+public class RecipeFilterCriteria {
+   private List<FilterRequest> filters;
+
+    public List<FilterRequest> getFilters() {
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeFilterCriteria.java:10]: I think you can use the @Getter @Setter adnotation

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeRepository.java [hunk 3]
```diff
@@ -44,6 +56,36 @@ public Recipe findOne(String s) {
         }
         return allRecipes;
     }
+    public List<Recipe> findBySteps(RecipeFilterCriteria recipeFilterCriteria) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeRepository.java:59]: Change the name of the function. findByFilter(...) might be a better name

File: iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeRepository.java [hunk 4]
```diff
@@ -44,6 +56,36 @@ public Recipe findOne(String s) {
...
+        TypedQuery<Recipe> typedQuery=entityManager.createQuery(criteriaQuery);
+        return typedQuery.getResultList();
+    }
+
+    private Predicate getPredicate(RecipeFilterCriteria recipeFilterCriteria, Root<Recipe> recipeRoot) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    List<Predicate> predicates=new ArrayList<>();
+    for(FilterRequest filterRequest:recipeFilterCriteria.getFilters()) {
+        if (filterRequest.getField().equals("title"))
+            predicates.add(criteriaBuilder.like(recipeRoot.get("title"), "%" + filterRequest.getText() + "%")
+            );
```
Comment (EnsyFane) [inline iChef-WebApi/src/main/java/com/kitchen/iChef/Repository/RecipeRepository.java:73]: fix formatting

**Review by EnsyFane — APPROVED** [review]

**Review by turcudaria — APPROVED** [review]

**Review by Stefpark77 — APPROVED** [review]

**Review by EnsyFane — APPROVED** [review]

**Pull request merged**